src/components/storage/CockpitStorageIntegration.jsx

```diff
@@ -509,7 +510,7 @@ const CheckStorageDialog = ({
     useEffect(() => {
         // If the required devices needed for manual partitioning are set up,
         // and prepare the partitioning
-        if (checkStep !== "prepare-partitioning") {
+        if (checkStep !== "prepare-partitioning" || appliedPartitioning) {
```

Comment:
Thinking out loud: do I understand it right that now that we reset partitioning on each rescan we need this added guard (non existing appliedPartitioning) to prevent preparing (applying) partitioning multiple times?

Reply:
Oh wait, I was looking at the patch in isolation, I guess I should look at the while first.

Reply:
Yes, if there is applied partitioning, we are sure that we are past this step.